src/libcore/str/mod.rs

```diff
@@ -3068,8 +3046,7 @@ impl str {
     /// The [pattern] can be a `&str`, [`char`], a slice of [`char`]s, or a
     /// function or closure that determines if a character matches.
     ///
-    /// [`char`]: primitive.char.html
-    /// [pattern]: str/pattern/index.html
+    /// [pattern]: self::pattern
```

Comment:
Hmm, this is not very clear that it's the module and not an associated function ... but I guess as long as the link works it's fine.

Reply:
It is, associated functions are `Self`